packages/react-core/src/components/Menu/MenuItem.tsx

```diff
@@ -290,49 +298,66 @@ const MenuItemBase: React.FunctionComponent<MenuItemProps> = ({
       role="none"
       {...props}
     >
-      <Component
-        tabIndex={-1}
-        className={css(styles.menuItem, getIsSelected() && styles.modifiers.selected, className)}
-        aria-current={getAriaCurrent()}
-        disabled={isDisabled}
-        role="menuitem"
-        ref={innerRef}
-        onClick={(event: any) => {
-          onItemSelect(event, onSelect);
-          _drill && _drill();
-        }}
-        {...additionalProps}
-      >
-        <span className={css(styles.menuItemMain)}>
-          {direction === 'up' && (
-            <span className={css(styles.menuItemToggleIcon)}>
-              <AngleLeftIcon aria-hidden />
-            </span>
-          )}
-          {icon && <span className={css(styles.menuItemIcon)}>{icon}</span>}
-          <span className={css(styles.menuItemText)}>{children}</span>
-          {isExternalLink && (
-            <span className={css(styles.menuItemExternalIcon)}>
-              <ExternalLinkAltIcon aria-hidden />
-            </span>
-          )}
-          {(flyoutMenu || direction === 'down') && (
-            <span className={css(styles.menuItemToggleIcon)}>
-              <AngleRightIcon aria-hidden />
-            </span>
-          )}
-          {getIsSelected() && (
-            <span className={css(styles.menuItemSelectIcon)}>
-              <CheckIcon aria-hidden />
+      <GenerateId>
+        {randomId => (
+          <Component
+            tabIndex={-1}
+            className={css(styles.menuItem, getIsSelected() && !hasCheck && styles.modifiers.selected, className)}
+            aria-current={getAriaCurrent()}
+            {...(!hasCheck && { disabled: isDisabled })}
+            role="menuitem"
+            ref={innerRef}
+            {...(!hasCheck && {
+              onClick: (event: any) => {
+                onItemSelect(event, onSelect);
+                _drill && _drill();
+              }
+            })}
+            {...(hasCheck && { htmlFor: randomId })}
+            {...additionalProps}
+          >
+            <span className={css(styles.menuItemMain)}>
+              {direction === 'up' && (
+                <span className={css(styles.menuItemToggleIcon)}>
+                  <AngleLeftIcon aria-hidden />
+                </span>
+              )}
+              {icon && <span className={css(styles.menuItemIcon)}>{icon}</span>}
+              {hasCheck && (
+                <span className={css('pf-c-menu__item-check')}>
+                  <Checkbox
```

Comment:
Ideally this should be a `<span>` because it is a child of a `<label>`, and [`<label>` only supports phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#technical_summary)

Doesn't look like the checkbox supports changing the element via `component` - could we add a follow up for that? https://github.com/patternfly/patternfly-react/blob/6e50f2379b8cd89bce6d5542a422def46162c3c4/packages/react-core/src/components/Checkbox/Checkbox.tsx#L96

Reply:
I opened issue #7433 and addressed it in this PR.